src/Maps/components/MapsContainer.ts

```diff
@@ -0,0 +1,242 @@
+import "../../ui/Maps.css";
+
+export interface MapsContainerState {
+    alertMessage?: string;
+    locations: Location[];
+    isFetchingData?: boolean;
+}
+
+export default class MapsContainer extends Component<MapsContainerProps, MapsContainerState> {
+    private subscriptionHandles: number[] = [];
+    readonly state: MapsContainerState = {
+        alertMessage: "",
+        locations: [],
+        isFetchingData: false
+    };
+
+    render() {
+        const mapsApiToken = this.props.apiToken ? this.props.apiToken.replace(/ /g, "") : undefined;
+        const commonProps = {
+            allLocations: this.state.locations,
+            fetchingData: this.state.isFetchingData,
+            className: this.props.class,
+            alertMessage: this.state.alertMessage,
+            divStyles: parseStyle(this.props.style),
+            onClickMarker: this.onClickMarker,
+            mapsToken: mapsApiToken,
+            ...this.props as MapProps
+        };
+
+        return this.props.mapProvider === "googleMaps"
+            ? createElement(googleApiWrapper, { ...commonProps })
+            : createElement(LeafletMap, { ...commonProps });
+    }
+
+    componentWillReceiveProps(nextProps: MapsContainerProps) {
+        this.resetSubscriptions(nextProps.mxObject);
+        const validationMessage = validateLocationProps(nextProps);
+        if (nextProps && nextProps.mxObject) {
+            if (validationMessage) {
+                this.setState({ alertMessage: validationMessage });
+            } else {
+                this.fetchData(nextProps.mxObject);
+            }
+        } else {
+            this.setState({ locations: [], alertMessage: "", isFetchingData: false });
+        }
+    }
+
+    componentWillUnmount() {
+        this.unsubscribeAll();
+    }
+
+    private resetSubscriptions(contextObject?: mendix.lib.MxObject) {
+        this.unsubscribeAll();
+        if (this.props.locations && this.props.locations.length) {
+            if (contextObject) {
+                this.subscriptionHandles.push(window.mx.data.subscribe({
+                    guid: contextObject.getGuid(),
+                    callback: () => this.fetchData(contextObject)
+                }));
+                this.props.locations.forEach(location => {
+                    this.subscriptionHandles.push(window.mx.data.subscribe({
+                        entity: location.locationsEntity as string,
+                        callback: () => this.fetchData(contextObject)
+                    }));
+                    [
+                        location.latitudeAttribute,
+                        location.longitudeAttribute,
+                        location.staticMarkerIcon,
+                        location.markerImageAttribute
+                    ]
+                    .forEach(
+                        (attr): number => this.subscriptionHandles.push(window.mx.data.subscribe({
```

Comment:
This is weird... `(attr): number =>` Don't need to be returning anything ... think `attr => this.` would be sufficient too.

Reply:
done